fml/mapping.h

```diff
@@ -125,6 +125,44 @@ class NonOwnedMapping final : public Mapping {
   FML_DISALLOW_COPY_AND_ASSIGN(NonOwnedMapping);
 };
 
+/// A Mapping like NonOwnedMapping, but uses Free as its release proc.
+class MallocMapping final : public Mapping {
+ public:
+  MallocMapping();
+
+  MallocMapping(const uint8_t* data, size_t size);
+
+  MallocMapping(fml::MallocMapping&& mapping);
+
+  ~MallocMapping() override;
+
+  /// Copies the data from `begin` to `end`.
+  /// It's templated since void* arithemetic isn't allowed and we want support
+  /// for `uint8_t` and `char`.
+  template <typename T>
+  static MallocMapping Copy(const T* begin, const T* end) {
```

Comment:
This needs some error handling to ensure we don't read OOB. Specifically, `begin > end` and `end > begin + size`. The return can be an MallocMapping.

Reply:
A suggestion: A variant I have found super useful in other projects is the following:
```
template <class T>
RetVal Copy(const T* buffer, size_t count) {
   return Copy(reinterpret_cast<const uint8_t*>(buffer), count * sizeof(buffer));
};
```

Which is an effective way of saying copy `count` objects of type `T` tightly packed in this buffer. If you want extra safeguards in place, you can even `std::enable_if` the specialization to ensure that T is `std::is_trivially_constructible`. That way, folks don't accidentally end up copying non-POD types.

Reply:
Done, I can't assert on size since it doesn't exist here.  The reason I chose this signature was just because it matched the code I was migrating from std::vector.  In the end I think I moved everything to the Copy signature with length, we could probably delete this but since it was helpful before I thought it might be helpful in the future.